android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt

```diff
@@ -338,6 +341,13 @@ fun Resource.appendRelatedEntityLocation(
     }
 }
 
+private fun updateReferenceList(
+  oldReferenceList: List<Reference>?,
+  newReference: Reference,
+): List<Reference> =
+  oldReferenceList.orEmpty().filterNot { it.reference.isNullOrEmpty() }.takeIf { it.isNotEmpty() }
```

Comment:
Is the `takeIf` function call needed? The previous filter has already excluded items that are null or empty.

Reply:
> Is the `takeIf` function call needed? The previous filter has already excluded items that are null or empty.

The `takeIf { it.isNotEmpty() } ?: listOf(newReference)` construct ensures that even if the `filterNot { it.reference.isNullOrEmpty() }` operation results in an empty list, we always return a valid reference list by appending the valid reference to an empty list.